### PR TITLE
Update CoreFx in preparation of ref harvesting

### PR DESCRIPTION
**Pull request by ericstj**
This PR fixes a couple bugs in XML and prepares the repo to pick up an updated buildtools that redistributes reference dlls in preference to the live build if possible.

I'll add the buildtools update to this PR if it is ready in time, otherwise I'll do it as a seperate PR.

I'd like to get this ready to ensure we don't have a place where buildtools updates are blocked.

/cc @weshaggard @joperezr

### CODE REVIEW

File: pkg/Microsoft.Private.PackageBaseline/packageIndex.json [hunk 1]
```diff
@@ -2012,7 +2008,7 @@
       "AssemblyVersionInPackageVersion": {
         "4.0.0.0": "4.0.0",
         "4.0.10.0": "4.0.10",
-        "4.1.0.0": "4.0.11",
+        "4.1.0.0": "4.3.0",
```
Comment (joperezr) [inline pkg/Microsoft.Private.PackageBaseline/packageIndex.json]: Is it by design that these guys are not sorted?

File: src/System.Xml.XPath.XmlDocument/src/System.Xml.XPath.XmlDocument.builds [hunk 2]
```diff
@@ -3,6 +3,9 @@
   <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.props))\dir.props" />
   <ItemGroup>
     <Project Include="System.Xml.XPath.XmlDocument.csproj" />
+    <Project Include="System.Xml.XPath.XmlDocument.csproj">
```
Comment (joperezr) [inline src/System.Xml.XPath.XmlDocument/src/System.Xml.XPath.XmlDocument.builds]: How come we didn't need this before? Was it shipped in box and is not anymore?

File: pkg/Microsoft.Private.PackageBaseline/packageIndex.json (hunk 1 above)
Comment (ericstj) [inline pkg/Microsoft.Private.PackageBaseline/packageIndex.json]: Yup, it's a dictionary.

File: src/System.Xml.XPath.XmlDocument/src/System.Xml.XPath.XmlDocument.builds (hunk 2 above)
Comment (ericstj) [inline src/System.Xml.XPath.XmlDocument/src/System.Xml.XPath.XmlDocument.builds]: Issue was introduced with recent XML refactoring.  This assembly contains a single type that isn't inbox in desktop.  In 1.0 this assembly built against public API which could be shared with desktop. Now it's building against System.Private.Xml which cannot be shared with desktop.

**Comment by ericstj** [comment]
OSX debug succeeded but failed when cleaning up due to IOException.
```
[xUnit] [ERROR] - The plugin hasn't been performed correctly: IOException on deletion
15:12:53 
[xUnit] [ERROR] - java.io.IOException: remote file operation failed: /Users/dotnet-bot/j/workspace/dotnet_corefx/master/osx_debug_prtest/generatedJUnitFiles/xUnit.Net at hudson.remoting.Channel@455b730e:dci-macpro-10: hudson.remoting.ChannelClosedException: channel is already closed
```

**Pull request merged**